numpy/ma/core.py

```diff
@@ -3802,7 +3802,10 @@ def __iadd__(self, other):
         else:
             if m is not nomask:
                 self._mask += m
-        ndarray.__iadd__(self._data, np.where(self._mask, 0, getdata(other)))
+        ndarray.__iadd__(
```

Comment:
Maybe not, but I do think the `self._data +=` form should do the same and be easier to read. Masked array is old, unmaintained, and has some subtleties that are easy to get wrong, but I don't think this is one of them. But maybe I'm missing something...